frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.tsx

```diff
@@ -408,11 +397,7 @@ export function FieldValuesWidgetInner({
     valuesMode,
   });
 
-  const isListMode =
-    !disableList &&
-    shouldList({ parameter, fields, disableSearch }) &&
-    valuesMode === "list" &&
-    !forceTokenField;
+  const isListMode = !disableList && valuesMode === "list" && !forceTokenField;
```

Comment:
We can remove `shouldList(...)` because it is implied by `valuesMode === "list"`. 

Reasoning follows: the only use of `setValuesMode` is here in `fetchValues`, setting `valuesMode` to "search".

This means that:
```
valuesMode === "list"
=> valuesMode === getValuesMode(...)  (that's how valuesMode is initialized)
=> shouldList({ parameter, fields, disableSearch }) === true
```

Reply:
didn't you remove `forceTokenField`?

Reply:
yes, separate PR. it's still valid

Reply:
I will fix conflicts

Reply:
fixed conflicts, forceTokenField is removed